src/gpu/intel/ocl/micro_sdpa.cl

```diff
@@ -249,6 +249,49 @@ micro_sdpa(const global KEY_DATA_T *K, const global half *Q,
             / VAL_ZP_ELEMENTS_PER_BYTE;
 #endif
 
+#ifdef PREFETCH_K0
+    /* Prefetch first K tile. */
+    cooperative_prefetch_2d_k(
+            /* ptr */ K,
+            /* r */ k,
+            /* c */ d, // faster than D_MAX
```

Comment:
It's not so much that it's faster than D_MAX but rather we need to avoid out-of-bounds prefetches.